### PR TITLE
Futures 03

### PR DESCRIPTION
**Pull request by nate-onesignal**
Futures-0.3 and Tokio-0.2 will be the maintained versions of these libraries going forward, and they have many speed improvements over the old ones.

I can squash these commits down into one larger one, if that would be better.

### CODE REVIEW

File: l337-postgres/src/lib.rs [hunk 1]
```diff
@@ -144,154 +141,127 @@ where
...
-                    .map(|connection| {
-                        sleep(Duration::from_secs(5));
-                        ((), connection)
-                    })
-                    .map_err(|e| l337::Error::External(e))
-            });
-
-            q1.join3(q2, q3)
-        });
+        let pool = Pool::new(mngr, config).await.unwrap();
```
Comment (DarrenTsung) [inline l337-postgres/src/lib.rs]: Do you know how this test proves that it reuses connections? I don't see anything functionally different from the previous implementation so this change is fine, just confused on what it was doing before

Comment (nate-onesignal) [reply l337-postgres/src/lib.rs]: Looking at it, I don't think this test proves that it reuses connections at all

**Comment by enzious** [comment]
I started having this issue which wasn't an issue on the previous l337 when using this branch
image

**Comment by enzious** [comment]
After using this branch, I've encountered an issue where I can't acquire a connection.

**Comment by nate-onesignal** [comment]
Thanks for reporting that, I will look into it

**Comment by nate-onesignal** [comment]
@enzious This is with l337-postgres?

**Comment by enzious** [comment]
Yep. I'm investigating more myself as well as I'm able. The problem emerged after migrating my codebase to await/async, so there is a possibility it's something to do with that but I don't see how that'd be the case as the Conn is dropping and I don't have any loops that'd hold it.

**Comment by nate-onesignal** [comment]
@enzious I found a deadlock, please update and try again

**Comment by enzious** [comment]
> @enzious I found a deadlock, please update and try again

It's working great! Thanks!

**Comment by enzious** [comment]
What are the pros of cons of using the std mutex over the tokio mutex?

**Pull request merged**